internal/compiler/typeloader.rs

```diff
@@ -261,6 +264,23 @@ impl TypeLoader {
         let referencing_file_or_url =
             import_token.and_then(|tok| tok.source_file().map(|s| s.path()));
 
+        if let Some(library) = maybe_relative_path_or_url.strip_prefix('@') {
+            let (library, file) = library
```

Comment:
Just a suggestion for distinguishing between the first variable that has the string before the tuple split, and the actual library name.

```suggestion
        if let Some(maybe_library_import) = maybe_relative_path_or_url.strip_prefix('@') {
            let (library, file) = maybe_library_import
```

That said, it's here where I would have expected the error handling to happen, instead of line 407. But I guess the heuristics are good enough.

Reply:
It would be nice to be able to handle errors here but I'm in trouble trying to pass a mutable `BuildDiagnostics` reference from `ensure_document_loaded()` because the state is already mutably borrowed there.

Reply:
What if the function returned a `Result` instead of an `Option`, since it's clearly an error? Just a thought :)